File: sorting/mergeSort.py
# arr = [4, 2, 5, 1, 9, 10, 0, 12, 11]
# def mergeArr(arr, L, R):
#   i=j=k=0
#   while i < len(L) and j < len(R):
#     if L[i] < R[j]:
#       arr[k] = L[i]
#       i += 1
#     else:
#       arr[k] = R[j]
#       j += 1
#     k += 1
#   while i < len(L):
#     arr[k] = L[i]
#     i += 1
#     k += 1
#   while j < len(R):
#     arr[k] = R[j]
#     j += 1
#     k += 1

  
# def mergeSort(arr):
#   if(len(arr) != 1):
#     mid = (0+len(arr))//2
#     L = arr[0:mid]
#     R = arr[mid:]
#     print(L, R, arr)
#     mergeSort(L)
#     mergeSort(R)
#     mergeArr(arr, L, R)
#     print(arr, 'test')
#   return arr
# print(mergeSort(arr))


arr = [1,3]
def check(a):
  a = 4
  

a = 1
check(a)
print(a)

File: js/mergesort.js
arr = [100, 34, 45, 2]
function mergeArr(L, R) {
const result = []
console.log(L, R, 'pp')
	var i = j = k = 0;
  while(i < L.length && j < R.length) {
  	if(L[i] < R[j]) {
    	result.push(L[i])
      i += 1
    } else {
      result.push(R[j])
     j += 1
    }
    k += 1
  }
  while(i < L.length) {
  	result.push(L[i])
    k += 1
    i += 1
  }
  while(j < R.length) {
  	 result.push(R[j])
    j += 1
    k += 1
  }
  console.log(result, 'res')
  return result
}
function mergeSort(arr) {
if(arr.length <= 1) return arr
	    let mid = parseInt(arr.length/2)
	    let L = mergeSort(arr.slice(0,mid))
	    let R = mergeSort( arr.slice(mid))
	    return mergeArr(L,R)

}

console.log(mergeSort(arr), 'check')
